embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/comms/api/ApiClientImpl.kt

```diff
@@ -84,16 +82,21 @@
      * server as a string.
      */
     private fun executeHttpRequest(connection: EmbraceConnection): ApiResponse<String> {
-        try {
+        return try {
             val responseCode = readHttpResponseCode(connection)
-            val headers = readHttpResponseHeaders(connection)
-            return ApiResponse(
-                responseCode,
-                headers,
-                readResponseBodyAsString(connection.inputStream)
-            )
+
+            val responseHeaders = readHttpResponseHeaders(connection)
+            return if (responseCode == HttpURLConnection.HTTP_OK) {
+                val responseBody = readResponseBodyAsString(connection.inputStream)
+                ApiResponse.Success(responseBody, responseHeaders)
+            } else {
+                val errorMessage = connection.errorStream?.let {
```

Comment:
I'm not sure if our server returns anything useful in the body. Is there a specific reason why you want to access this now when we didn't use it before?